chasm/lib/activity/statemachine.go

```diff
@@ -25,20 +36,48 @@ func (a *Activity) SetStateMachineState(state activitypb.ActivityExecutionStatus
 var TransitionScheduled = chasm.NewTransition(
 	[]activitypb.ActivityExecutionStatus{
 		activitypb.ACTIVITY_EXECUTION_STATUS_UNSPECIFIED,
+		activitypb.ACTIVITY_EXECUTION_STATUS_STARTED, // On retries for start-to-close timeouts
 	},
 	activitypb.ACTIVITY_EXECUTION_STATUS_SCHEDULED,
-	func(a *Activity, ctx chasm.MutableContext, _ any) error {
+	func(a *Activity, ctx chasm.MutableContext, scheduledEvent scheduledEvent) error {
 		attempt, err := a.Attempt.Get(ctx)
 		if err != nil {
 			return err
 		}
 
+		currentTime := time.Now()
```

Comment:
Use `ctx.Now(a)` so we can later support time skipping and advanced pause features.